CONTENTS.md

```diff
@@ -0,0 +1,423 @@
+* A set of [Tampermonkey](https://www.tampermonkey.net) scripts (see [userscript](userscript)
+  directory) to extend OBS features when using the web interface.
+* Several [systemd](systemd) units, as many of the tools (e.g., the `staging-bot`) make use of
+  them.
+
+## Tools
+
+Most of these tools are available as packages for several distributions. Check the [spec file in
+this repository](dist/package/openSUSE-release-tools.spec) or the [devel
+project](https://build.opensuse.org/package/show/openSUSE:Tools/openSUSE-release-tools) for further
+information.
+
+For the time being, we have classified them into three different groups: *command line tools*, *OBS
+bots* and *osc plugins*. Bear in mind that the information in the following list might be wrong and
+incomplete.
+
+### Command Line Tools
+
+ Usually, the executables are renamed as `osrt-NAME` (e.g., `osrt-announcer`).
+
+#### announcer
+
+Generates email diffs summaries to announce product releases.
+
+* Sources: [factory-package-news/announcer.py](factory-package-news/announcer.py)
+* Documentation: [factory-package-news/README.asciidoc](factory-package-news/README.asciidoc)
+* Package: openSUSE-release-tools-announcer
+
+#### pkglistgen
+
+Generates and updates OBS products for openSUSE and SLE. It generates package lists based on
+`000package-groups` and puts them in `000product` (resulting kiwi files) and `000release-packages`
+(release package spec files).
+
+* Sources: [pkglistgen.py](pkglistgen.py)
+* Documentation: [docs/pkglistgen.md](docs/pkglistgen.md)
+* Package: openSUSE-release-tools-pkglistgen
+* Usage: gocd
+
+#### container_cleaner
+
+Cleans up the `maintenance_release` projects for containers
+
+* Sources: [container_cleaner.py](container_cleaner.py)
+* Documentation: --
+* Package: openSUSE-release-tools
+* Usage: gocd
+
+#### metrics
+
+Generates insightful metrics from relevant OBS and annotation data, based on InfluxDB and Grafana.
+
+* Sources: [metrics.py](metrics.py)
+* Documentation: [docs/metrics.md](./docs/metrics.md)
+* Package: openSUSE-release-tools-metrics
+* Usage: ?
+
+#### metrics-access
+
+Ingests `download.opensuse.org` Apache access logs and generate metrics. It is composed of a PHP.
+script and a set of [systemd units](systemd).
+
+* Sources: [metrics/access/aggregate.php](metrics/access/aggregate.php)
+* Documentation: [docs/metrics.md](./docs/metrics.md)
+* Package: openSUSE-release-tools-metrics-access
+* Usage: ?
+
+#### totest-manager
+
+Releases distribution snapshots to openQA and publishes if the result is positive.
+
+* Sources: [totest-manager.py](totest-manager.py) and [ttm](ttm)
+* Documentation: [ttm/README.md](ttm/README.md)
+* Package: openSUSE-release-tools
+* Usage: gocd
+
+#### unmaintained
+
+Finds unmaintained binaries sourced from SLE.
+
+* Sources: [unmaintained.py](unmaintained.py)
+* Documentation: --
+* Package: openSUSE-release-tools
+* Usage: ?
+
+#### sync-rebuild
+
+Syncs openSUSE:Factory and openSUSE:Factory:Rebuild.
```

Comment:
Maybe exaplanation of purpose of Rebuild subproject can help here to see a bit bigger picture why it is useful. ( I expect it is used when rebuild of factory is needed like with new gcc? )

Reply:
@coolo please, are @jreidinger assumptions right? From the code, I see that it triggers the rebuild of packages that failed to build in `openSUSE:Factory` and `openSUSE:Factory:Rebuild`. When and why is it used?

Reply:
am not @coolo though :)

AFAIK `sync-rebuild` isn't running as a standalone service and hasn't be called by others. The code was merged to `accept command` long time ago https://github.com/openSUSE/openSUSE-release-tools/commit/87c891662015f14421c2315210c248e712e697c8 

* Basically `openSUSE:Factory` and `openSUSE:Factory:Rebuild` should have the same build fails, if not, re-trigger it to rebuild either in `openSUSE:Factory` or `openSUSE:Factory:Rebuild`
* The major difference between `openSUSE:Factory` and `openSUSE:Factory:Rebuild` is that `openSUSE:Factory` uses `rebuild=local` as its rebuild strategy
* The *rebuild a package* function will be triggered when a new Factory checkin runs, ie. accepting an(or multiple) staging project

Reply:
Thanks a lot, @nilxam! I guess I need to learn a little bit more about rebuilding strategies (local, transitive and direct) to get the whole point :smiley:. The reason is documented in the referenced commit: *Trigger rebuild of packages that failed build in either openSUSE:Factory or openSUSE:Factory:Rebuild, but not the other. Helps over the fact that openSUSE:Factory uses rebuild=local, thus sometimes 'hiding' build failures.*

So I would say this one is obsolete too, right?

Reply:
yes, you can mark it as obsoleted.